pyk/src/pyk/ktool/krun.py

```diff
@@ -246,7 +257,14 @@ def _krun(
         else:
             bug_report.add_command(args)
 
-    return run_process(args, check=check, pipe_stderr=pipe_stderr, logger=logger or _LOGGER, exec_process=debugger)
+    return run_process(
+        args,
+        check=check,
+        pipe_stderr=pipe_stderr,
+        pipe_stdout=not proof_hint,
```

Comment:
Consider exposing `pipe_stdout` instead, and checking `not proof_hint or pipe_stdout`. You can raise `ValueError` with a descriptive error message to tell the user why this is necessary.